lib/xgu/xgu.h

```diff
@@ -712,6 +779,99 @@ uint32_t* xgu_set_light_local_attenuation(uint32_t* p, unsigned int light_index,
     return push_floats(p, v.f, 3);
 }
 
+/* ==== Texture stuff ==== */
```

Comment:
Major issue: All of these are arrays which control 4 different texture units; so all of these will need another parameter afte `p`: `uint32_t texture_index`

The command would respectively be `... + texture_index * <stride>` - the stride is probably different per command?